pkg/codegen/schema/loader.go

```diff
@@ -258,16 +223,25 @@ func LoadPackageReferenceV2(
 		return nil, err
 	}
 
-	if descriptor.Name != ref.Name() ||
-		descriptor.Version != nil &&
+	name := descriptor.Name
+	if descriptor.Parameterization != nil {
+		name = descriptor.Parameterization.Name
+	}
+	version := descriptor.Version
+	if descriptor.Parameterization != nil {
+		version = &descriptor.Parameterization.Version
+	}
+
+	if name != ref.Name() ||
+		version != nil &&
 			ref.Version() != nil &&
-			!ref.Version().Equals(*descriptor.Version) {
-		if l, ok := loader.(*pluginLoader); ok {
-			return nil, fmt.Errorf("req: %s@%v: entries: %v (returned %s@%v)", descriptor.Name, descriptor.Version,
+			!ref.Version().Equals(*version) {
+		if l, ok := loader.(*cachedLoader); ok {
+			return nil, fmt.Errorf("req: %s@%v: entries: %v (returned %s@%v)", name, version,
```

Comment:
I'm fine with this but equally wonder if we should just add `Stringer` or the like to loaders and delegate to that polymorphically

Reply:
I guess we only want this when we have a cache because the other loaders wouldn't have anything interesting to say.